crates/anvil-polkadot/tests/it/genesis.rs

```diff
@@ -0,0 +1,44 @@
+use crate::utils::{TestNode, assert_with_tolerance, unwrap_response};
+use alloy_primitives::U256;
+use anvil_core::eth::EthRequest;
+use anvil_polkadot::config::{AnvilNodeConfig, SubstrateNodeConfig};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_genesis() {
+    let genesis_block_number: u32 = 1000;
+    let anvil_genesis_timestamp: u64 = 42;
+    let chain_id: u64 = 4242;
+    let anvil_node_config = AnvilNodeConfig::test_config()
+        .with_genesis_block_number(Some(genesis_block_number))
+        .with_genesis_timestamp(Some(anvil_genesis_timestamp))
+        .with_chain_id(Some(chain_id));
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config, substrate_node_config).await.unwrap();
+
+    // Check that block number, timestamp, and chain id are set correctly at genesis
+    assert_eq!(node.best_block_number().await, genesis_block_number);
+    let genesis_hash = node.block_hash_by_number(genesis_block_number).await.unwrap();
+    // Anvil genesis timestamp is in seconds, while Substrate timestamp is in milliseconds.
+    let genesis_timestamp = anvil_genesis_timestamp.checked_mul(1000).unwrap();
+    let actual_genesis_timestamp = node.get_decoded_timestamp(Some(genesis_hash)).await;
+    assert_eq!(actual_genesis_timestamp, genesis_timestamp);
+    let current_chain_id = node.get_decoded_chain_id(Some(genesis_hash)).await;
+    assert_eq!(current_chain_id, chain_id);
+
+    // Manually mine two blocks and force the timestamp to be increasing with 1 second each time.
+    unwrap_response::<()>(
```

Comment:
The timestamp of the second block will include everything that happens from when the TimeManager is created until the inherent is created, so maybe take a larger tolerance in the assert